frontend/packages/pipelines-plugin/src/components/repository/types.ts

```diff
@@ -0,0 +1,12 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { PipelineRunStatus } from '../../types';
+
+export type RepositoryKind = K8sResourceKind & {
+  spec: {
+    url: string;
+    branch?: string;
+    event_type?: string;
+    namespace?: string;
+  };
+  pipelinerun_status: PipelineRunStatus[];
```

Comment:
Optional field, this will not be available when you have a new Repository.
```suggestion
  pipelinerun_status?: PipelineRunStatus[];
```
`PipelineRunStatus` is not correct type.  I believe this is regular pipelineRun status, for Repository its completely different. Please craft a new type  `RepositoryPipelineRunStatus`

Reply:
crafted a new type